src/theory/bags/bags_rewriter.cpp

```diff
@@ -71,6 +71,7 @@ RewriteResponse BagsRewriter::postRewrite(TNode n)
       case BAG_CHOOSE: response = rewriteChoose(n); break;
       case BAG_CARD: response = rewriteCard(n); break;
       case BAG_IS_SINGLETON: response = rewriteIsSingleton(n); break;
+      case BAG_FROM_SET: response = rewriteFromSet(n); break;
```

Comment:
What about BAG_TO_SET?  You should add a rewrite function and I believe there is a dual rewrite rule (bag.to_set (mk_bag x n)) = (singleton x).

Reply:
I thought about this. But if `n` is a variable, then it can be assigned a negative or zero multiplicity later, which would make the bag empty. As a result, bag.to_set` returns the emptyset. If `n` is a positive constant, then, this is not in a normal form, and  should be handled by `NormalForm::evaluate` which would return `(singleton (singleton_op t) x)`.

Reply:
Ah right. Although I think there is still a case where the element is not constant and the multiplicity is a postitive constant, where you'd need e.g.
`(bag.to_set (mkBag x 4)) ---> (singleton x)`.

Reply:
Done. I missed this case when the element is a variable. Thanks for pointing this out.